user.py

```diff
@@ -16,15 +32,70 @@ def __init__(self, username: str, hashed_password: str):
 class UserDataService:
 
     def load(self):
+        self.__ensure_database_exists()
         try:
-            users = open(os.getenv("PSG_USER_DATABASE_PATH"), "rb")
+            users = open(users_info_path, "rb")
             all_users = pickle.load(users)
             users.close()
             return all_users
         except Exception as e:
             print(str(e))
             return {}
 
+    def save_user(self, user: User):
+        self.__ensure_database_exists()
+        try:
+            users = open(users_info_path, "rb")
+            all_users = pickle.load(users)
+            users.close()
+
+            all_users[user.username] = user.hashed_password
+            users = open(users_info_path, "wb")
+            pickle.dump(all_users, users)
+            users.close()
+
+        except Exception as error:
+            logger.error("database creation failed: " + str(error))
+            raise UserDataServiceException("Sorry but we cannot currently access the database.")
+
+    def check_if_user_unique(self, new_user):
+        self.__ensure_database_exists()
+        try:
+            users = open(users_info_path, "rb")
+            all_users = pickle.load(users)
+            if new_user in all_users:
```

Comment:
Since you're returning a boolean depending on the result of the if-condition, you can omit the if-condition completely, and just `return new_user in all_users`.

Reply:
Fixed.